src/metatrain/utils/augmentation.py

```diff
@@ -118,17 +135,85 @@ def _apply_wigner_D_matrices(
     target_tmap: TensorMap,
     transformations: List[torch.Tensor],
     wigner_D_matrices: Dict[int, List[torch.Tensor]],
+    target_type: str,
 ) -> TensorMap:
+    """
+    For the given target ``target_tmap``, for each of its blocks, splits the values by
+    system and applies a SO(3) transformation by action of a Wigner-D matrix.
+    """
+    # Iterate over blocks in the target
     new_blocks: List[TensorBlock] = []
     for key, block in target_tmap.items():
         ell, sigma = int(key[0]), int(key[1])
         values = block.values
-        if "atom" in block.samples.names:
-            split_values = torch.split(
-                values, [len(system.positions) for system in systems]
-            )
+
+        # First, build the indices that split the block samples by system
+        split_indices: List[int] = []
+
+        if target_type == "spherical":
```

Comment:
what's the difference between `target_type == "spherical"` and `target_type == "atomic_basis_spherical"`?

Reply:
atomic basis spherical doesn't have all the atomic samples in a given block - only those with the corresponding to the atom types. The splitting along the samples axis needs more care than for pure spherical targets